.github/workflows/updpksums.yml

```diff
@@ -0,0 +1,188 @@
+            )
+            const fs = require('fs')
+            const path = require('path')
+            const packages = files.map(e => path.dirname(e.filename)).reduce((a, directory) => {
+              if (a[0] !== directory && fs.existsSync(path.join(process.env.WORKTREE_PATH, directory, 'PKGBUILD'))) {
+                if (directory.indexOf(' ') >= 0) throw new Error(`${directory}: contains spaces!`)
+                a.unshift(directory)
+              }
+              return a
+            }, [])
+            return packages.join(' ')
+      - name: Identify actor
+        if: steps.determine-packages.outputs.result != ''
+        id: actor
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const githubApiRequest = require('./github-api-request')
+            const answer = await githubApiRequest(
+              console,
+              '${{ steps.setup.outputs.token }}',
+              'GET',
+              `/users/${process.env.ACTOR}`
+            )
+            core.setOutput('login', answer.login)
+            core.setOutput('name', answer.name)
+            core.setOutput('email', answer.email || `${process.env.ACTOR}@users.noreply.github.com`)
+      - name: Configure build
+        if: steps.determine-packages.outputs.result != ''
+        shell: bash
+        run: |
+          USER_NAME="${{ steps.actor.outputs.name }}" &&
+          USER_EMAIL="${{ steps.actor.outputs.email }}" &&
+          mkdir -p "$HOME" &&
+          git config --global user.name "$USER_NAME" &&
+          git config --global user.email "$USER_EMAIL" &&
+          echo "PACKAGER=$USER_NAME <$USER_EMAIL>" >>$GITHUB_ENV
+      - name: Update packages
+        if: steps.determine-packages.outputs.result != ''
+        id: update
+        shell: bash
+        env:
+          PACKAGES: ${{ steps.determine-packages.outputs.result }}
+        run: |
+          for PACKAGE_TO_UPGRADE in $PACKAGES
+          do
+            cd "/usr/src/$REPO/$PACKAGE_TO_UPGRADE" &&
+            update_script="$GITHUB_WORKSPACE/update-scripts/checksums/$PACKAGE_TO_UPGRADE" &&
+            if test -f "$update_script"
+            then
+              $update_script "$UPGRADE_TO_VERSION"
+            else
+              updpkgsums
+            fi &&
+
+            # give `pkgver()` a chance
+            if grep '^pkgver *()' PKGBUILD
+            then
+              # give makepkg a chance to update the version
+              git fetch --unshallow "https://github.com/$OWNER/$REPO" "refs/pull/$PR_NUMBER/head" &&
+              makepkg --noextract --nobuild
+            fi ||
+            exit 1
+          done
+
+          git update-index -q --refresh &&
+          if git diff-files --exit-code
+          then
+            echo "::notice::No checksums needed to be updated in $PACKAGES"
+            exit 0
+          fi &&
+
+          msg="$PACKAGES: update checksums" &&
+          git commit -sm "$msg" PKGBUILD &&
```

Comment:
we're looping through 1 or more packages above, couldn't that result in multiple PKGBUILD files needing to be commited?

Reply:
Right you are! I squashed this:

```diff
diff --git a/.github/workflows/updpksums.yml b/.github/workflows/updpksums.yml
index 3a83f35..340ae90 100644
--- a/.github/workflows/updpksums.yml
+++ b/.github/workflows/updpksums.yml
@@ -162,7 +162,7 @@ jobs:
           fi &&
 
           msg="$PACKAGES: update checksums" &&
-          git commit -sm "$msg" PKGBUILD &&
+          git commit -asm "$msg" &&
           echo "msg=$msg" >>$GITHUB_OUTPUT &&
           echo "modified=true" >>$GITHUB_OUTPUT
       - name: Determine target repository
```